policy/diamond/policy/admin/admin.rego

```diff
@@ -2,11 +2,14 @@ package diamond.policy.admin
 
 import rego.v1
 
-is_admin(subject) if {
-	"super_admin" in data.diamond.data.subjects[subject].permissions # regal ignore:external-reference
-}
+is_admin[subject] := "super_admin" in data.diamond.data.subjects[subject].permissions
 
-is_beamline_admin(subject, beamline) if {
-	some admin in data.diamond.data.subjects[subject].permissions
-	beamline in data.diamond.data.admin[admin] # regal ignore:external-reference
+is_beamline_admin[subject] contains beamline if {
+	some subject
+	some role in data.diamond.data.subjects[subject].permissions
+	some beamline in data.diamond.data.admin[role]
 }
```

Comment:
Not sure this name really makes sense given it maps to a list of beamline names. We could rename it or use a double mapping like:

```suggestion
is_beamline_admin[subject][beamline] := true if {
	some permission in data.diamond.data.subjects[subject].permissions
        some beamline in data.diamond.data.admin[permissions]
}
```

Reply:
Okay, after a bit of messing around:

```suggestion
is_beamline_admin[subject][beamline] := true if {
    some permission in data.diamond.data.subjects[subject].permissions
    some beamline in data.diamond.data.admin[permissions]
}

is_beamline_admin[subject][beamline] := false if {
    data.diamond.data.subjects[subject]
    data.diamond.data.beamlines[beamline]
    some permission in data.diamond.data.subjects[subject].permissions
    some _beamline in data.diamond.data.admin[permissions]
    beamline != _beamline
}```

Reply:
That only includes beamlines where the user is an admin so the value is always true at which point we might as well use a set again. Rename it to something like `beamlines_for_user`? I can't think of a better name so open to suggestions

Reply:
Yeah, feel like it would be better to go with something like the latter. `beamlines_for_subject` or `subject_beamlines` would make sense if we were to keep it as a list.

On another note, we should probably avoid calling them `beamlines` and prefer `instruments` instead as it includes microscopes etc

Reply:
I've renamed it. I'll start using `instrument` when the data bundle stops listing `beamline` for each session and the `beamlines` subtree is renamed.

Reply:
Maybe it should be `beamline_admin_for_subject` instead. It's a bit unclear otherwise whether it's a list of beamlines a subjects is on sessions for.